src/test/java/core/basesyntax/OperationHandlerTest.java

```diff
@@ -0,0 +1,112 @@
+package core.basesyntax;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.exception.ProductNotFoundException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.impl.operation.BalanceOperationHandler;
+import core.basesyntax.strategy.impl.operation.PurchaseOperationHandler;
+import core.basesyntax.strategy.impl.operation.ReturnOperationHandler;
+import core.basesyntax.strategy.impl.operation.SupplyOperationHandler;
+import core.basesyntax.strorage.FruitStorage;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class OperationHandlerTest {
+    private static OperationHandler balanceHandler;
+    private static OperationHandler purchaseHandler;
+    private static OperationHandler returnHandler;
+    private static OperationHandler supplyHandler;
+
+    @BeforeClass
+    public static void setUp() {
+        balanceHandler = new BalanceOperationHandler();
+        purchaseHandler = new PurchaseOperationHandler();
+        returnHandler = new ReturnOperationHandler();
+        supplyHandler = new SupplyOperationHandler();
+    }
+
+    @Before
+    public void clearStorage() {
+        FruitStorage.fruits.clear();
+    }
+
+    @Test
+    public void apply_balanceOperation_ok() {
+        balanceHandler.apply(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                "banana", 20));
+        balanceHandler.apply(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                "apple", 80));
+        int actualApple = FruitStorage.fruits.get("apple");
+        int expectedBanana = 20;
+        int expectedApple = 80;
+        int actualBanana = FruitStorage.fruits.get("banana");
+        assertEquals("Not valid banana quantity", actualBanana, expectedBanana);
+        assertEquals("Not valid apple quantity",actualApple, expectedApple);
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
Try to locate all negative tests (notOk) after positive(ok). It will be more readable, I think